src/Form/PolicyForm.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Core\Config\Entity\ConfigEntityStorageInterface;
+use Drupal\Core\Entity\EntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\user\RoleStorageInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Password policy config entity form.
+ */
+class PolicyForm extends EntityForm {
+
+  /**
+   * Config entity storage.
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface
+   */
+  protected $policyConfigEntityStorage;
+
+  /**
+   * Role storage.
+   *
+   * @var \Drupal\user\RoleStorageInterface
+   */
+  protected $roleStorage;
+
+  /**
+   * Constructs the password policy config entity form.
+   *
+   * @param \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $policy_config_entity_storage
+   *   Password policy config entity storage.
+   * @param \Drupal\user\RoleStorageInterface $role_storage
+   *   Role storage.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Messenger.
+   */
+  public function __construct(ConfigEntityStorageInterface $policy_config_entity_storage, RoleStorageInterface $role_storage, MessengerInterface $messenger) {
+    $this->policyConfigEntityStorage = $policy_config_entity_storage;
+    $this->roleStorage = $role_storage;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): PolicyForm {
+    return new static(
+      $container->get('entity_type.manager')->getStorage('password_enhancements_policy'),
+      $container->get('entity_type.manager')->getStorage('user_role'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+    $form = parent::form($form, $form_state);
+
+    $form['#title'] = $this->t('Edit policy');
+
+    $form['name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Name'),
+      '#default_value' => $this->entity->getName(),
+      '#required' => TRUE,
+    ];
+
+    $options = [];
+    foreach ($this->roleStorage->loadMultiple() as $role) {
```

Comment:
Consider using `array_map()` that does not pollute the global scope with remnant variables ;)

Reply:
Also, filter out roles that already has saved a policy.

Reply:
The `array_map` is way slower than `foreach` as it is calling a function constantly, what about unsetting the variables after the `foreach` instead?

Reply:
Actually it is a loop, it's calling a given function for every element: https://www.php.net/manual/en/function.array-map.php